packages/intl/src/GenericTranslator.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Tempest\Intl;
+
+use Tempest\EventBus\EventBus;
+use Tempest\Intl\Catalog\Catalog;
+use Tempest\Intl\IntlConfig;
+use Tempest\Intl\Locale;
+use Tempest\Intl\MessageFormat\Formatter\MessageFormatter;
+
+final readonly class GenericTranslator implements Translator
+{
+    public function __construct(
+        private(set) IntlConfig $config,
+        private(set) Catalog $catalog,
+        private MessageFormatter $formatter,
+        private ?EventBus $eventBus = null,
+    ) {}
+
+    public function translateForLocale(Locale $locale, string $key, mixed ...$arguments): string
+    {
+        if (! $this->catalog->has($locale, $key)) {
+            $this->eventBus?->dispatch(new TranslationMiss(
```

Comment:
Out of curiosity: on translation miss, we dispatch an event, and the resulting string will simply be the untranslated string with its placeholders. I assume we go the event route instead of throwing an exception because we don't want to hard-error whenever a translation is missing.

I don't think we do anything with the event yet, did you plan on adding some kind of debugging tool/log later that collects missing translations?